src/Tutor.Core/LearnerModel/DomainOverlay/KnowledgeComponentMasteries/KnowledgeComponentMastery.cs

```diff
@@ -231,30 +191,32 @@ private void When(SessionAbandoned @event)
 
         private void When(AssessmentItemAnswered @event)
         {
-            /* Probably refactor to move the GetMaximumSubmissionCorrectness from AE to KCMastery, or a 
-             * child object (which would be created here if it doesn't exist yet). The Apply method
-             * should never fail, silently or otherwise, and fetching the AE can fail. */
-            var assessmentItem = KnowledgeComponent.GetAssessmentItem(@event.Submission.AssessmentItemId);
-            if (assessmentItem == null) return;
+            var itemId = @event.Submission.AssessmentItemId;
+            var assessmentMastery = AssessmentMasteries.Find(am => am.AssessmentItemId == itemId);
+            if (assessmentMastery == null)
+                throw new InvalidOperationException("No assessment mastery for item: " + itemId +". Were the masteries created and loaded correctly?");
 
-            var currentCorrectnessLevel = assessmentItem.GetMaximumSubmissionCorrectness();
+            assessmentMastery.LastSubmissionTime = @event.TimeStamp;
+            if (assessmentMastery.Mastery > @event.Submission.CorrectnessLevel) return;
 
-            assessmentItem.Submissions.Add(@event.Submission);
+            assessmentMastery.UpdateMastery(@event.Submission.CorrectnessLevel);
+            UpdateMastery();
+        }
```

Comment:
We still have a bit of a problem here with where exceptions happen. Since this method doesn't run until the event was already recorded as valid, and it would be used for rehydrating the aggregate (if we ever needed to load the aggregate from events), it should theoretically never fail. So, if there are any exceptions, they need to happen in SubmitAssessmentItemAnswer instead. 
But even if SubmitAssessmentItemAnswer checks that the AssessmentItemMastery exists, this method obviously needs to find it again, and finding an object in a list is an operation that can fail anyway (maybe not really in practice, but theoretically it can). So I'm really not sure how to handle this in a way that is actually formally in line with Event Sourcing conventions, other than having the When method create the AIMastery if it doesn't exist instead of throwing an exception.